### PR TITLE
🎉 New Source: Punk API [low-code cdk]

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Developing new connector for the source: Punk API*
*New data source will be definitely useful for destinations seeking training data*

Resolves:
[🎉 New Source: Punk API](https://github.com/airbytehq/connector-contest/issues/231)

## How
*Developed using low-code cdk*

## Recommended reading order
1. `spec.yaml`
2. `punk-api.yaml`
3. `schemas/*`

## 🚨 User Impact 🚨
- No breaking changes, Just addition of new source
<details><summary><strong>New Connector: Source Punk API</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

</details>

## Tests

<details><summary><strong>Integration & Acceptance</strong></summary>
Full Test Results

image

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello @btkcodedev, Marcos from Airbyte here 👋 . We received more than 25 new contributions along the weekend. One is yours 🎉  thank so much for! Our team is limited and maybe the review process can take longer than expected. As described in the Airbyte's Hacktoberfest your contribution was submitted before November 2nd and it is eligible to win the prize. The review process will validate other requirements. I ask to you patience until someone from the team review it. 

Because I reviewed some contributions for Hacktoberfest so far I saw some common patterns you can check in advance:
* [ ] Make sure you have added connector documentation to `/docs/integrations/`
* [ ] Remove the file `catalog` from `/integration_tests`
* [ ] Edit the `sample_config.json` inside `/integration_tests`
* [ ] For the `configured_catalog` you can use only `json_schema: {}`
* [ ] Add title to all properties in the `spec.yaml`
* [ ] Make sure the `documentationUrl` in the `spec.yaml` redirect to Airbyte's future connector page, eg: connector Airtable the `documentationUrl: https://docs.airbyte.com/integrations/sources/airtable`
* [ ] Review now new line at EOF (end-of-file) for all files.

If possible send to me a DM in Slack with the tests credentials, this process will make easier to us run integration tests and publish your connector. If you only has production keys, make sure to create a bootstrap.md explaining how to get the keys.

**Comment by marcosmarxm** [comment]
@btkcodedev please request my review when you solved all points I commented in my previous comment.

**Comment by marcosmarxm** [comment]
Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.

Sorry the inconvenience and see you again next week, thank you so much for your contribution!

**Comment by btkcodedev** [comment]
@sh4sh @marcosmarxm All comments resolved.
Community Review Required :)

**Comment by btkcodedev** [comment]
@sh4sh Also please let me know whether any further changes are required.

File: airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "beers"
+      path: "/beers"
+
+  beers_with_id_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "beers_with_id"
+      path: "/beers/{{ config['id'] or '1'}}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml:36]: Please correct this, use request_parameter option to inject this values.

File: airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      field_pointer: []
+  requester:
+    url_base: "https://api.punkapi.com/v2/"
+    http_method: "GET"
+
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml:15]: The pagination is missing.

File: airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    requester:
+      $ref: "*ref(definitions.requester)"
+
+  base_stream:
+    schema_loader:
+      type: JsonSchema
+      file_path: "./source_punk_api/schemas/{{ options['name'] }}.json"
+    retriever:
+      $ref: "*ref(definitions.retriever)"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml:24]: The API allows using before and after parameters, this allow to implement incremental syncs. Please check the DateStreamSlicer component.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

File: airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml (hunk 1 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml]: Done

File: airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml (hunk 2 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-punk-api/source_punk_api/punk_api.yaml:41]: Done with separate stream.

**Comment by marcosmarxm** [comment]
@btkcodedev I'll review your PR today.

**Review by marcosmarxm — APPROVED** [review]
Thanks @btkcodedev

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**